File: snapshots/input/syntax/Main/Statements.cs
using System.Diagnostics.CodeAnalysis;

namespace Main;

[SuppressMessage("ReSharper", "all")]
public class Statements
{
    void Try()
    {
        try
        {
            File.ReadLines("asd");
        }
        catch (Exception err)
        {
            Console.WriteLine(err);
        }
    }

    (string a, bool b) Default()
    {
        (string a, bool b) c = default;
        return c;
    }

    void Deconstruction()
    {
        var point = (1, 2);
        int x = 0;
        (x, int y) = point;
    }

    record Inferred(int F1, int F2);

    void InferredTuples()
    {
        var list = new List<Inferred>();
        var result = list.Select(c => (c.F1, c.F2)).Where(t => t.F2 == 1);
    }

    int MultipleInitializers()
    {
        int a = 1, b = 2;
        return a + b;
    }

    void RefVariable()
    {
        var data = new int[] { 1, 2 };
        ref var value = ref data[0];
    }

    class MyDisposable : IDisposable
    {
        public void Dispose()
        {
        }
    }

    MyDisposable Using()
    {
        var b = new MyDisposable();
        using (var a = b)
        {
            return a;
        }
    }

    long MultipleUsing()
    {
        using (Stream a = File.OpenRead("a"), b = File.OpenRead("a"))
        {
            return a.Length + b.Length;
        }
    }

    int Foreach()
    {
        var y = new int[] { 1 };
        var z = 0;
        foreach (int x in y)
            z += x;
        return z;
    }
}

File: snapshots/input/syntax/VBMain/Statements.vb
Imports System.Diagnostics.CodeAnalysis
Imports System.IO

Namespace VBMain
    <SuppressMessage("ReSharper", "all")>
    Public Class Statements
        Private Sub [Try]()
            Try
                File.ReadLines("asd")
            Catch err As Exception
                Console.WriteLine(err)
            End Try
        End Sub

        Private Function [Default]() As (A As String, B As Boolean)
            Dim C As (A As String, B As Boolean) = ("42", 42)
            Return C
        End Function

        Public Class Inferred
            Property F1 As Int32
            Property F2 As Int32
        End Class

        Private Sub InferredTuples()
            Dim List = New List(Of Inferred)()
            Dim Result = List.Select(Function(c) (c.F1, c.F2)).Where(Function(t) t.F2 = 1)
        End Sub

        Private Function MultipleInitializers() As Integer
            Dim a As Integer = 1, b As Integer = 2
            Return a + b
        End Function

        Class MyDisposable
            Implements IDisposable

            Private Sub Dispose() Implements IDisposable.Dispose
                Throw New NotImplementedException()
            End Sub
        End Class

        Private Function [Using]() As MyDisposable
            Dim b = New MyDisposable()

            Using a = b
                Return a
            End Using
        End Function

        Private Function MultipleUsing() As Long
            Using a As Stream = File.OpenRead("a"), b As Stream = File.OpenRead("a")
                Return a.Length + b.Length
            End Using
        End Function

        Private Function Foreach() As Integer
            Dim y = New Integer() {1}
            Dim z = 0

            For Each x As Integer In y
                z += x
            Next

            Return z
        End Function

    End Class
End Namespace
